backend/routes/rankings.js

```diff
@@ -0,0 +1,65 @@
+import express from "express";
+import { Follow } from "../models/follow.js";
+import { Stock } from "../models/stock.js";
+
+const router = express.Router();
+
+const checkSession = (req, res, next) => {
```

Comment:
No need to repeat this function everywhere. Let's have a common one, and then import it and "use it" `router.use()` where needed.